internal/shim-sgx/src/handler/mod.rs

```diff
@@ -126,59 +266,118 @@ impl<'a> Handler<'a> {
             Some(ExceptionVector::Page) => {
                 h.print_ssa_stack_trace();
                 h.gdb_session();
-                h.exit(1)
+                let _ = h.exit_group(1);
+                unreachable!()
             }
 
             _ => h.attacked(),
         }
     }
 
+    fn get_attestation(
+        &mut self,
+        hash: usize,
+        hash_len: usize,
+        buf: usize,
+        buf_len: usize,
+    ) -> Result<[usize; 2], libc::c_int> {
+        if hash_len == 0 || buf_len == 0 {
+            return Ok([QUOTE_SIZE, TECH]);
+        }
+        let hash = {
+            let h = self.validate_slice::<u8>(hash, hash_len)?;
```

Comment:
Shouldn't we check for `null` here as well, especially before this line?

Reply:
Shouldn't `validate_slice` error out then?